### PR TITLE
ROU-3217: OnCellValueChangeEvent of Data Grid not fired when two columns have the same binding

### PR DESCRIPTION
**Pull request by JoaoFerreira-FrontEnd**
This PR is for OnCellValueChangeEvent of Data Grid to be fired even when two columns have the same binding

### What was happening
* The OnCellValueChangeEvent was not being fired when I have two columns in the data grid binding to the same attribute in the database. I set a client action as the handler to the event on one of the columns. When I changed the value of the cell of that column, the client action does not get executed. However, when I remove the other column with the same binding, the event is fired and handled by the client action as it should be.

### What was done
* Change _triggerEventsFormColumn to use column index instead of column binding as input parameter

### Test Steps
1. Add a DataGrid to the screen
2. Config multiple columns with the same binding; 
3. Add custom events to some of the columns on OnCellValueChange;
4. Check if the event feedback is the expected

### Checklist
* [x] tested locally
* [x] documented the code
* [x] clean all warnings and errors of eslint
* [ ] requires changes in OutSystems (if so, provide a module with changes)
* [ ] requires new sample page in OutSystems (if so, provide a module with changes)

### CODE REVIEW

File: code/src/WijmoProvider/Features/DirtyMark.ts [hunk 1]
```diff
@@ -3,8 +3,7 @@ namespace WijmoProvider.Feature {
     export class DirtyMark
         implements
             OSFramework.Feature.IDirtyMark,
-            OSFramework.Interface.IBuilder
-    {
```
Comment (gnbm) [inline code/src/WijmoProvider/Features/DirtyMark.ts:7]: @JoaoFerreira-FrontEnd The automation in Github are failing. Please check the feedback

**Review by gnbm — CHANGES_REQUESTED** [review]

**Review by joselrio — APPROVED** [review]

File: code/src/WijmoProvider/Features/ValidationMark.ts [hunk 2]
```diff
@@ -38,19 +38,27 @@ namespace WijmoProvider.Feature {
...
             );
             // The old value can be captured on the dirtyMark feature as it is the one responsible for saving the original values
             const oldValue = this._grid.features.dirtyMark.getOldValue(
                 e.row,
-                binding
+                column.binding
+            );
+            this._triggerEventsFromColumn(
+                e.row,
+                column.index,
```
Comment (glundgren93) [inline code/src/WijmoProvider/Features/ValidationMark.ts:58]: We should be passing column uniqueId instead of column index.

**Review by glundgren93 — CHANGES_REQUESTED** [review]

File: code/src/WijmoProvider/Features/ValidationMark.ts [hunk 3]
```diff
@@ -304,7 +315,7 @@ namespace WijmoProvider.Feature {
                 );
                 this._triggerEventsFromColumn(
                     action.row,
-                    binding,
+                    action.col,
```
Comment (glundgren93) [inline code/src/WijmoProvider/Features/ValidationMark.ts:318]: We should be passing column uniqueId instead of column index.

**Review by glundgren93 — CHANGES_REQUESTED** [review]
Since we changed the behavior of the events, we should guarantee that it didnt cause any side effects as well.

File: code/src/WijmoProvider/Features/ValidationMark.ts (hunk 2 above)
Comment (gnbm) [reply code/src/WijmoProvider/Features/ValidationMark.ts]: @JoaoFerreira-FrontEnd This is still missing, right?

File: code/src/WijmoProvider/Features/ValidationMark.ts (hunk 3 above)
Comment (gnbm) [reply code/src/WijmoProvider/Features/ValidationMark.ts]: @JoaoFerreira-FrontEnd This is still missing, right?

**Review by gnbm — CHANGES_REQUESTED** [review]

File: code/src/WijmoProvider/Features/ValidationMark.ts (hunk 3 above)
Comment (JoaoFerreira-FrontEnd) [reply code/src/WijmoProvider/Features/ValidationMark.ts]: No, this is done

**Review by gnbm — DISMISSED** [review]

File: code/src/WijmoProvider/Features/ValidationMark.ts [hunk 4]
```diff
@@ -229,19 +241,22 @@ namespace WijmoProvider.Feature {
...
             rowNumber: number,
-            binding: string,
+            columnUniqueID: string,
             // eslint-disable-next-line @typescript-eslint/no-explicit-any
             oldValue: any,
             // eslint-disable-next-line @typescript-eslint/no-explicit-any
             newValue: any
         ) {
-            const column = this._grid.getColumn(binding);
+            const column = this._grid
```
Comment (glundgren93) [inline code/src/WijmoProvider/Features/ValidationMark.ts:256]: We should use: 
const column = this._grid.getColumn(columnUniqueID);

**Review by glundgren93 — CHANGES_REQUESTED** [review]

**Review by gnbm — DISMISSED** [review]

File: code/src/WijmoProvider/Features/ValidationMark.ts (hunk 4 above)
Comment (JoaoFerreira-FrontEnd) [reply code/src/WijmoProvider/Features/ValidationMark.ts]: done

File: code/src/WijmoProvider/Features/ValidationMark.ts [hunk 5]
```diff
@@ -638,7 +654,7 @@ namespace WijmoProvider.Feature {
             // Triggers the events of OnCellValueChange associated to a specific column in OS
             this._triggerEventsFromColumn(
                 rowNumber,
-                column.provider.binding,
+                column.provider.uniqueId,
```
Comment (glundgren93) [inline code/src/WijmoProvider/Features/ValidationMark.ts:657]: This should be column.uniqueId. The uniqueId is our property, not the providers.

File: code/src/WijmoProvider/Features/ValidationMark.ts [hunk 6]
```diff
@@ -667,7 +683,7 @@ namespace WijmoProvider.Feature {
                         // Triggers the events of OnCellValueChange associated to a specific column in OS
                         this._triggerEventsFromColumn(
                             rowNumber,
-                            column.provider.binding,
+                            column.provider.uniqueId,
```
Comment (glundgren93) [inline code/src/WijmoProvider/Features/ValidationMark.ts:686]: This should be column.uniqueId. The uniqueId is our property, not the providers.

**Review by glundgren93 — CHANGES_REQUESTED** [review]

**Review by glundgren93 — DISMISSED** [review]

**Review by glundgren93 — APPROVED** [review]

**Review by joselrio — APPROVED** [review]

**Pull request merged**